File: 3461. Check If Digits Are Equal in String After Operations I.cpp
class Solution {
public:
    bool hasSameDigits(string s) {
        while (s.size() > 2) {
            string t;
            for (int i=0; i+1<s.size(); ++i)
                t.push_back('0' + (s[i]-'0'+s[i+1]-'0')%10);
            s = t;
        }
        return s[0] == s[1];
    }
};

File: 3461. Check If Digits Are Equal in String After Operations I.py
class Solution:
    def hasSameDigits(self, ss: str) -> bool:
        s = list(ss)
        while len(s) > 2:
            s = [str((int(s[i])+int(s[i+1])) % 10) for i in range(0, len(s)-1)]
        return s[0] == s[1]